tools/integration-tests/tester/tests/diagnostics/diagnostics_test.go

```diff
@@ -96,3 +102,42 @@ func TestDiagnosticApis(t *testing.T) {
 	require.NoError(t, err, "error while reading tools/diagnostic/drng csv")
 	require.Equal(t, drngHeader, records[0], "unexpected drngHeader header")
 }
+
+func TestSendMessageAPI(t *testing.T) {
+	ctx, cancel := tests.Context(context.Background(), t)
+	defer cancel()
+	n, err := f.CreateNetwork(ctx, t.Name(), 1, framework.CreateNetworkConfig{
+		StartSynced: true,
+	})
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(ctx, t, n)
+	node := n.Peers()[0]
+	api := node.GoShimmerAPI
+
+	sent := tests.SendDataMessages(t, n.Peers()[:1], 5)
+	parentMessageIDs := tangle.NewParentMessageIDs()
+
+	for k := range sent {
+		parentID, _ := tangle.NewMessageID(k)
+		parentMessageIDs = parentMessageIDs.Add(tangle.StrongParentType, parentID)
+	}
+	msg, err := tangle.NewMessage(
```

Comment:
Message is prepared anyway by the node we send the request to, it is a little misleading that we need to create this dummy message and put the bytes. What about passing only parents and payload to api.SendMessage() and this wrapping and changing to bytes will be done by client lib

Reply:
Great idea actually. We should have the API only accept parents and payload. :)